compiler/src/main/kotlin/io/spine/protodata/renderer/SourceFile.kt

```diff
@@ -75,7 +76,19 @@ private constructor(
     /**
      * Writes the source code into the file on the file system.
      */
-    public fun write(charset: Charset = Charsets.UTF_8) {
-        path.writeText(code, charset, WRITE, TRUNCATE_EXISTING)
+    internal fun write(charset: Charset = Charsets.UTF_8, rootDir: Path? = null) {
+        val targetPath = if (rootDir != null) {
```

Comment:
How about having an extension function for `Path` which does what this `if` does?